### PR TITLE
Feature add event queue

### PR DESCRIPTION
**Pull request by heavenshell**
## Problem
I added async feature before. But currently, handling huge file is too slow.
This is because sending huge buffer to TSServer every time.

So, I add debunce feature.

This PR is useful for handling huge file.

### Step to reproduce

1. Add following setting to vimrc.
```viml
autocmd InsertLeave,TextChanged,BufWritePost *.ts,*.tsx call tsuquyomi#asyncGeterr()
```
2. Open https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts
3. Press `x` few seconds

kapture 2019-01-19 at 23 42 53

## Solution
This idea is from [vim-lsp](https://github.com/prabirshrestha/vim-lsp/pull/237) and thanks to @mattn 

Apply this patch and setting.
```viml
autocmd InsertLeave,TextChanged,BufWritePost *.ts,*.tsx call tsuquyomi#asyncGeterr(1000)
```

kapture 2019-01-19 at 23 39 26

### CODE REVIEW

File: autoload/tsuquyomi/tsClient.vim [hunk 1]
```diff
@@ -396,6 +399,7 @@ endfunction
 " PARAM: {string} cmd Command type. e.g. 'completion', etc...
 " PARAM: {dictionary} args Arguments object. e.g. {'file': 'myApp.ts'}.
 function! tsuquyomi#tsClient#sendCommandAsyncEvents(cmd, args)
+  let s:quickfix_list = []
```
Comment (heavenshell) [inline autoload/tsuquyomi/tsClient.vim]: Clear quickfix before send event.

File: autoload/tsuquyomi.vim [hunk 2]
```diff
@@ -109,6 +109,38 @@ function! s:setqflist(quickfix_list, ...)
   endif
 endfunction
 
+let s:diagnostics_queue = []
+let s:diagnostics_timer = -1
+function! s:addDiagnosticsQueue(delay, bufnum)
+  if index(s:diagnostics_queue, a:bufnum) != -1
```
Comment (heavenshell) [inline autoload/tsuquyomi.vim]: Check same bufnum is contained in queue.
If exists, do nothing.

**Comment by Quramy** [comment]
I've checked your delayed async get errors function via `:call tsuquyomi#asyncGeterr(5000)` and it works pretty well 😄

**Comment by Quramy** [comment]
@heavenshell Sorry for my late review and reply :bow: and thanks for your great contribution !

**Comment by heavenshell** [comment]
Thank you so much!!

**Pull request merged**